Test the model's ability to find the lead vocalist of a South African band

richardwooding opened the pull request:
# Thank you for contributing an eval! ♥️

🚨 Please make sure your PR follows these guidelines, __failure to follow the guidelines below will result in the PR being closed automatically__. Note that even if the criteria are met, that does not guarantee the PR will be merged nor GPT-4 access granted. 🚨

__PLEASE READ THIS__:

In order for a PR to be merged, it must fail on GPT-4. We are aware that right now, users do not have access, so you will not be able to tell if the eval fails or not. Please run your eval with GPT-3.5-Turbo, but keep in mind as we run the eval, if GPT-4 gets higher than 90% on the eval, we will likely reject since GPT-4 is already capable of completing the task.

We plan to roll out a way for users submitting evals to see the eval performance on GPT-4 soon. Stay tuned! Until then, you will not be able to see the eval performance on GPT-4. **Starting April 10, the minimum eval count is 15 samples, we hope this makes it easier to create and contribute evals.**

## Eval details 📑
### Eval name
south-african-bands

### Eval description

Test the model's ability to understand that we are providing the name of a South African band, find the supplied band, and if the band has a lead vocalist provide the stage name or real name of the vocalist. All bands provided have one lead vocalist.

### What makes this a useful eval?

Test whether GPT-4 can distinguish South African band names, can find their lead vocalist, can distinguish whether
the artist uses a stage name or not, and preferentially return a stage name over a real name.

## Criteria for a good eval ✅

Below are some of the criteria we look for in a good eval. In general, we are seeking cases where the model does not do a good job despite being capable of generating a good response (note that there are some things large language models cannot do, so those would not make good evals).

Your eval should be:

- [X] Thematically consistent: The eval should be thematically consistent. We'd like to see a number of prompts all demonstrating some particular failure mode. For example, we can  create an eval on cases where the model fails to reason about the physical world.
- [X] Contains failures where a human can do the task, but either GPT-4 or GPT-3.5-Turbo could not.
- [X] Includes good signal around what is the right behavior. This means either a correct answer for `Basic` evals or the `Fact` Model-graded eval, or an exhaustive rubric for evaluating answers for the `Criteria` Model-graded eval.
- [X] **Include at least 15 high quality examples.**

If there is anything else that makes your eval worth including, please document it below.

### Unique eval value

> Insert what makes your eval high quality that was not mentioned above. (Not required)

## Eval structure 🏗️

Your eval should
- [X] Check that your data is in `evals/registry/data/{name}`
- [X] Check that your yaml is registered at `evals/registry/evals/{name}.yaml`
- [X] Ensure you have the right to use the data you submit via this eval

(For now, we will only be approving evals that use one of the existing eval classes. You may still write custom eval classes for your own cases, and we may consider merging them in the future.)

## Final checklist 👀

### Submission agreement

By contributing to Evals, you are agreeing to make your evaluation logic and data under the same MIT license as this repository. You must have adequate rights to upload any data used in an Eval. OpenAI reserves the right to use this data in future service improvements to our product. Contributions to OpenAI Evals will be subject to our usual Usage Policies (https://platform.openai.com/docs/usage-policies).

- [X] I agree that my submission will be made available under an MIT license and complies with OpenAI's usage policies.

### Email address validation

If your submission is accepted, we will be granting GPT-4 access to a limited number of contributors. Access will be given to the email address associated with the merged pull request.

- [X] I acknowledge that GPT-4 access will only be granted, if applicable, to the email address used for my merged pull request.

### Limited availability acknowledgement

We know that you might be excited to contribute to OpenAI's mission, help improve our models, and gain access to GPT-4. However, due to the requirements mentioned above and high volume of submissions, we will not be able to accept all submissions and thus not grant everyone who opens a PR GPT-4 access. We know this is disappointing, but we hope to set the right expectation before you open this PR.

- [X] I understand that opening a PR, even if it meets the requirements above, does not guarantee the PR will be merged nor GPT-4 access granted.

### Submit eval

- [X] I have filled out all required fields in the evals PR form
- [X] (Ignore if not submitting code) I have run `pip install pre-commit; pre-commit install` and have verified that `black`, `isort`, and `autoflake` are running when I commit and push

Failure to fill out all required fields will result in the PR being closed.

### Eval JSON data 

Since we are using Git LFS, we are asking eval submitters to add in as many Eval Samples (at least 5) from their contribution here:

<details>
  <summary>View evals in JSON</summary>

  ### Eval
  ```jsonl
  {"input": [{"role": "system", "content": "A user will give you a name of a South African band. Name the lead vocalist, if they use a stage name answer with that instead of real name. Restrict answers to name only, with no further explanations being necessary."},{"role": "user", "content":  "Little Sister"}], "ideal":  "Maria de Fatima"}
{"input": [{"role": "system", "content": "A user will give you a name of a South African band. Name the lead vocalist, if they use a stage name answer with that instead of real name. Restrict answers to name only, with no further explanations being necessary."},{"role": "user", "content":  "Prime Circle"}], "ideal":  "Ross Learmonth"}
{"input": [{"role": "system", "content": "A user will give you a name of a South African band. Name the lead vocalist, if they use a stage name answer with that instead of real name. Restrict answers to name only, with no further explanations being necessary."},{"role": "user", "content":  "Four Jacks and a Jill"}], "ideal":  "Glenys Lynne"}
{"input": [{"role": "system", "content": "A user will give you a name of a South African band. Name the lead vocalist, if they use a stage name answer with that instead of real name. Restrict answers to name only, with no further explanations being necessary."},{"role": "user", "content":  "Rabbitt"}], "ideal":  "Trevor Rabin"}
{"input": [{"role": "system", "content": "A user will give you a name of a South African band. Name the lead vocalist, if they use a stage name answer with that instead of real name. Restrict answers to name only, with no further explanations being necessary."},{"role": "user", "content":  "The Narrow"}], "ideal":  "Hanu de Jong"}
{"input": [{"role": "system", "content": "A user will give you a name of a South African band. Name the lead vocalist, if they use a stage name answer with that instead of real name. Restrict answers to name only, with no further explanations being necessary."},{"role": "user", "content":  "Just Jinjer"}], "ideal":  "Ard Matthews"}
  ```
</details>

Ein-Tim commented:
The PR itself looks good to me, but its title could be shorter :)

richardwooding commented:
I shorted the title of the PR @Ein-Tim

usama-openai reviewed (changes requested):
Thanks for the contribution. This eval looks good, but the sample file has a few issues. It seems like the ideal answer for some queries is not correct. For example:

- For the band "Little Sister" the ideal answer is "Maria de Fatima" but the correct answer seems to be "Debbi Lonmon"
- For the band "The Skylarks" the ideal answer is "Dolly Rathebe" but the correct answer seems to be "Miriam Makeba"

Can you kindly validate that all the ideal answers in the dataset are correct?

We would love to review the PR again after the suggested changes.

richardwooding commented:
I've gone through the dataset manually, removed some bands I found where the data was questionable, and added 5 more which I double checked. Please can you consider my updated dataset.

usama-openai commented:
You need to merge the `master` branch into your branch to resolve workflow-related issues. Kindly update your branch with the latest master branch.

richardwooding commented:
Thank you I merged main as you requested. 

How do I make sure this reflects against my 
OpenAI organizational account: richard.wooding@spandigital.com <-- correct one
And not my personal account: richard.wooding@gmail.com

Both are linked to my github account, and I thought using git configure with my email would work.

I'd appreciated your advice @usama-openai

usama-openai reviewed (approved):
This PR looks in good shape now. I'm approving this PR.
